### ISSUE TITLE
JVP: Option to Disable Gradient Caching for Tangents

### ISSUE THREAD

**Issue by qsh-zh** [pytorch/pytorch#151782]
### 🚀 The feature, motivation and pitch

I'm requesting a new option for `torch.func.jvp` to disable gradient caching and tracking specifically for the tangent output without affecting the primal output.

Currently, when using `torch.func.jvp(fn, primals, tangents)`, the JVP output requires gradients by default, which causes it to cache activations unnecessarily. In many use-cases like mine, the JVP tangent vectors are used for auxiliary calculations but are not part of the computation graph for backpropagation. This leads to:

1. Unnecessary memory usage from cached activations for the tangent calculations
2. No way to selectively disable gradient tracking for just the tangent part

I propose adding an optional parameter to `torch.func.jvp`, perhaps named `track_tangent_grad=True` (defaulting to True for backward compatibility), that would allow users to disable gradient tracking specifically for the JVP output without affecting the primal output and without requiring multiple forward passes.

Example of desired usage:
```python
# Current behavior
output, jvp = func.jvp(layer, (x, ), (v, ))
assert output.requires_grad == True
assert jvp.requires_grad == True  # Caches activations unnecessarily

# Proposed behavior
output, jvp = func.jvp(layer, (x, ), (v, ), track_tangent_grad=False)
assert output.requires_grad == True
assert jvp.requires_grad == False  # No activation caching
```

### Alternatives

Currently, I have to use workarounds like:
1. Detaching the JVP output after calculation (`jvp.detach()`), which works but is less efficient than not caching in the first place
2. Using separate forward passes (`layer(x)` and then a separate JVP calculation with `no_grad`), which increases computation
3. Complex wrappers around the JVP function, which reduce code clarity

None of these solutions are ideal, as they either increase computation or don't fully prevent the unnecessary caching during the forward pass.

**Comment by ryushinn** [comment]
Also looking forward to this. Here I would also like to pitch two user cases that can greatly benefit from this feature: [Mean Flow](https://arxiv.org/pdf/2505.13447) and [sCM](https://arxiv.org/abs/2410.11081). Their training involves `jvp` computation but do `stopgrad` upon the tangent. (See Alg.1 in Mean Flow).
